src/app/shared/directives/toggle-password.directive.ts

```diff
@@ -0,0 +1,29 @@
+import { Directive, ElementRef, Inject } from '@angular/core';
+import { DOCUMENT } from '@angular/common'; 
+
+@Directive({
+  selector: '[appTogglePassword]'
+})
+export class TogglePasswordDirective {
+  private _shown = false;
+
+  constructor(private el: ElementRef) {
+    const parent = this.el.nativeElement.parentNode;
+    const span = document.createElement('i');
+    span.setAttribute('class','bi bi-eye-slash-fill')
```

Comment:
To make some style classes available for all app, you can put them into `src/styles.scss`. For more details https://angular.io/guide/component-styles#external-and-global-style-files